### ISSUE TITLE
Prisma Client: `disconnect: true` does not appear to delete the foreign key in the returned data

### ISSUE THREAD

**Issue by casey-chow** [prisma/prisma#20491]
In Prisma 5.1.0 (Postgres) I noticed that my tests began failing to properly execute with:

```
      device = await prisma.device.update({
        where: { id: device.id },
        data: { user: { disconnect: true } },
      });
```

The device returned is still connected to the user after this is run.

**Comment by janpio** [comment]
That is concerning @casey-chow.

Can you please turn on logging in your project when running one of these queries? (Instructions see https://pris.ly/d/logging) It would be very interesting to see the query that is being executed under the hood here. If you can downgrade to Prisma 5.0.0 as well and confirm that it correctly works there, and what query you get with that version, that would be amazing. Thank you!

To complete the information our bug issue template usually asks for:
What OS and Node version are you running? Can you maybe shared your full `prisma -v` output? Thanks.

**Comment by Jolg42** [comment]
@casey-chow Along with filling more information mentioned in our bug report template, sharing your Prisma schema would also help to reproduce this.

**Comment by Jolg42** [comment]
I could reproduce something, is that what you meant @casey-chow ?
```ts
import { PrismaClient } from "@prisma/client";

async function main() {
  const prisma = new PrismaClient();

  // Before (should be empty)
  const usersBefore = await prisma.user.findMany();
  console.log({ usersBefore });
  const devicesBefore = await prisma.device.findMany();
  console.log({ devicesBefore });

  const newDeviceWithUser = await prisma.device.create({
    data: {
      user: {
        create: {},
      },
    },
  });
  console.log({ newDeviceWithUser });

  // Disconnect!
  const disconnect = await prisma.device.update({
    where: { id: newDeviceWithUser.id },
    data: { user: { disconnect: true } },
  });
  console.log({ disconnect });

  // After
  const usersAfter = await prisma.user.findMany();
  console.log({ usersAfter });
  const devicesAfter = await prisma.device.findMany();
  console.log({ devicesAfter });

  // Cleanup
  await prisma.device.deleteMany();
  await prisma.user.deleteMany();

  prisma.$disconnect();
}

main();

// 5.0.0
// { usersBefore: [] }
// { devicesBefore: [] }
// { newDeviceWithUser: { id: 1, userId: 8 } }
// { disconnect: { id: 1, userId: null } }
// { usersAfter: [ { id: 8, name: null } ] }
// { devicesAfter: [ { id: 1, userId: null } ] }

// 5.1.0
// { usersBefore: [] }
// { devicesBefore: [] }
// { newDeviceWithUser: { id: 2, userId: 9 } }
// { disconnect: { id: 2, userId: 9 } } ---- HERE! `userId` should be null
// { usersAfter: [ { id: 9, name: null } ] }
// { devicesAfter: [ { id: 2, userId: null } ] }
```

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id     Int      @id @default(autoincrement())
  name   String?
  Device Device[]
}

model Device {
  id     Int   @id @default(autoincrement())
  user   User? @relation(fields: [userId], references: [id])
  userId Int?
}
```

**Comment by Jolg42** [comment]
Note that we are looking at this and might issue a patch release, we'll keep you updated, thanks for the report 🙌🏼

**Comment by Jolg42** [comment]
Note: we got a fix, more info coming soon.

**Comment by casey-chow** [comment]
What a flurry to wake up to! Apologies for not following bug template, I figured there should be one but for whatever reason it didn't ever pop up for me in GitHub, and it was near the end of the day and wanted to get this information over to your team quickly. 

Yes, I was able to verify it worked in 5.0.0 as well. Thanks for getting this fix in so quickly, look forward to seeing the patch release!
